tests/run-make/llvm-outputs/rmake.rs

```diff
@@ -0,0 +1,20 @@
+// test that directories get created when emitting llvm bitcode and IR
+
+use run_make_support::{cwd, run_in_tmpdir, rustc};
+use std::path::PathBuf;
+
+fn main() {
+    let mut path_bc = PathBuf::new();
+    let mut path_ir = PathBuf::new();
+    run_in_tmpdir(|| {
+        let p = cwd();
+        path_bc = p.join("nonexistant_dir_bc");
+        path_ir = p.join("nonexistant_dir_ir");
+        rustc().input("-").stdin("fn main() {}").out_dir(&path_bc).arg("--emit=llvm-bc").run();
+        rustc().input("-").stdin("fn main() {}").out_dir(&path_ir).arg("--emit=llvm-ir").run();
+        assert!(path_bc.exists());
+        assert!(path_ir.exists());
+    });
+    assert!(!path_bc.exists());
+    assert!(!path_ir.exists());
```

Comment:
Suggestion: these asserts should not be necessary because the directories and output artifacts will be removed when `run_in_tmpdir` returns.

Reply:
does the testing framework have unit tests?

generally i think it's not a very good idea for tests to just assume everything is working.

Reply:
I will add a self-test for the `run_in_tmpdir` helper in a follow-up PR, but in my opinion ordinary tests should not need to check for that.

Reply:
Adding a self test in https://github.com/rust-lang/rust/pull/126223